File: VECTOAux/VectoAuxiliaries/Hvac/DeleteCell.vb
Imports System.Windows.Forms
Imports System.Drawing


Public Class DeleteCell
	Inherits DataGridViewButtonCell

	Public Property ToolTip As String = "Delete tech benefit line"
	Private del As Image = TryCast(My.Resources.ResourceManager.GetObject("Delete"), Image)


	Protected Overrides Sub Paint(graphics As Graphics, clipBounds As Rectangle, cellBounds As Rectangle,
								rowIndex As Integer, elementState As DataGridViewElementStates, value As Object, formattedValue As Object,
								errorText As String, cellStyle As DataGridViewCellStyle, advancedBorderStyle As DataGridViewAdvancedBorderStyle,
								paintParts As DataGridViewPaintParts)

		advancedBorderStyle.All = DataGridViewAdvancedCellBorderStyle.Single

		Me.ToolTipText = ToolTip

		cellStyle.BackColor = Color.White
		MyBase.Paint(graphics, clipBounds, cellBounds, rowIndex, elementState, value, formattedValue, errorText, cellStyle,
					advancedBorderStyle, paintParts)
		graphics.DrawImage(del, cellBounds)
	End Sub
End Class

Public Class DeleteAlternatorCell
	Inherits DataGridViewButtonCell

	Public Property ToolTip As String = "Delete alternator"
	Private del As Image = TryCast(My.Resources.ResourceManager.GetObject("Delete"), Image)


	Protected Overrides Sub Paint(graphics As Graphics, clipBounds As Rectangle, cellBounds As Rectangle,
								rowIndex As Integer, elementState As DataGridViewElementStates, value As Object, formattedValue As Object,
								errorText As String, cellStyle As DataGridViewCellStyle, advancedBorderStyle As DataGridViewAdvancedBorderStyle,
								paintParts As DataGridViewPaintParts)

		advancedBorderStyle.All = DataGridViewAdvancedCellBorderStyle.Single

		Me.ToolTipText = ToolTip

		cellStyle.BackColor = Color.White
		MyBase.Paint(graphics, clipBounds, cellBounds, rowIndex, elementState, value, formattedValue, errorText, cellStyle,
					advancedBorderStyle, paintParts)
		graphics.DrawImage(del, cellBounds)
	End Sub
End Class

File: VECTOAux/BusAuxiliaries/DownstreamModules/Impl/HVAC/DeleteCell.cs
using System.Drawing;
using System.Reflection;
using System.Windows.Forms;

namespace TUGraz.VectoCore.BusAuxiliaries.DownstreamModules.Impl.HVAC {
	public class DeleteCell : DataGridViewButtonCell
	{
		public string ToolTip { get; set; } = "Delete tech benefit line";
		private Image del = null;  //= My.Resources.ResourceManager.GetObject("Delete") as Image;


		protected override void Paint(
			Graphics graphics, Rectangle clipBounds, Rectangle cellBounds, int rowIndex, DataGridViewElementStates elementState,
			object value, object formattedValue, string errorText, DataGridViewCellStyle cellStyle,
			DataGridViewAdvancedBorderStyle advancedBorderStyle, DataGridViewPaintParts paintParts)
		{
			if (del == null) {
				var assembly = Assembly.GetExecutingAssembly();
				var file = assembly.GetManifestResourceStream("Delete");
				del = Image.FromStream(file);
			}
			advancedBorderStyle.All = DataGridViewAdvancedCellBorderStyle.Single;

			this.ToolTipText = ToolTip;

			cellStyle.BackColor = Color.White;
			base.Paint(
				graphics, clipBounds, cellBounds, rowIndex, elementState, value, formattedValue, errorText, cellStyle,
				advancedBorderStyle, paintParts);
			graphics.DrawImage(del, cellBounds);
		}
	}

	public class DeleteAlternatorCell : DataGridViewButtonCell
	{
		public string ToolTip { get; set; } = "Delete alternator";

		protected Image del = null;
		//private Image del = My.Resources.ResourceManager.GetObject("Delete") as Image;

		protected override void Paint(
			Graphics graphics, Rectangle clipBounds, Rectangle cellBounds, int rowIndex, DataGridViewElementStates elementState,
			object value, object formattedValue, string errorText, DataGridViewCellStyle cellStyle,
			DataGridViewAdvancedBorderStyle advancedBorderStyle, DataGridViewPaintParts paintParts)
		{
			if (del == null) {
				var assembly = Assembly.GetExecutingAssembly();
				var file = assembly.GetManifestResourceStream("Delete");
				del = Image.FromStream(file); 
			}


			advancedBorderStyle.All = DataGridViewAdvancedCellBorderStyle.Single;

			this.ToolTipText = ToolTip;

			cellStyle.BackColor = Color.White;
			base.Paint(
				graphics, clipBounds, cellBounds, rowIndex, elementState, value, formattedValue, errorText, cellStyle,
				advancedBorderStyle, paintParts);
			graphics.DrawImage(del, cellBounds);
		}
	}
}